base/TimeStringConversion.F90

```diff
@@ -239,4 +240,22 @@ function string_to_esmf_timeinterval(time_interval_string,unusable,rc) result(ti
 
    end function string_to_esmf_timeinterval
 
+   function hconfig_to_esmf_timeinterval(hconfig, key, unusable, rc) result(time_interval)
```

Comment:
Hopefully some day, this will be a single call to ESMF, which then can just be inserted where needed?

Reply:
@tclune, ESMF has the ability to take an an ISO duration in a string and turn it into an ESMF_TimeInterval. It is just that this is in not in a release yet. As soon we have a release with this we can get, then yes we can use this, until just have to parse it ourselves. I even made an issue to remind us to do this: https://github.com/GEOS-ESM/MAPL/issues/2736